skills/dff_knowledge_prompted_skill/scenario/response.py

```diff
@@ -0,0 +1,196 @@
+    )
+
+    shared_memory = int_ctx.get_shared_memory(ctx, actor)
+    prompt = shared_memory.get("prompt", "")
+    prompt = PROMPT
+
+    custom_el = (
+        ctx.misc.get("agent", {})
+        .get("dialog", {})
+        .get("human_utterances", [{}])[-1]
+        .get("annotations", {})
+        .get("custom_entity_linking")
+    )
+    user_kg = (
+        ctx.misc.get("agent", {})
+        .get("dialog", {})
+        .get("human_utterances", [{}])[-1]
+        .get("annotations", {})
+        .get("user_knowledge_memorizer")
+    )
+    logger.info(f"custom_el: {custom_el}")
+    logger.info(f"user_kg: {user_kg}")
+
+    if USE_KG_DATA and user_kg and (kg_prompt := user_kg["kg_prompt"]):
+        kg_prompt = re.sub(r"[-\n]", "", kg_prompt[0].lower()).split(".")
+        kg_prompt = ",".join(kg_prompt)
+        prompt = prompt + f"\n\nADDITIONAL INSTRUCTION: You know that {kg_prompt}. Use these facts in your answer."
+
+    logger.info(f"prompt: {prompt}")
+    logger.info(f"dialog_context: {dialog_context}")
+
+    if len(dialog_context) > 0:
+        try:
+            hypotheses = send_request_to_prompted_generative_service(
+                dialog_context,
+                prompt,
+                GENERATIVE_SERVICE_URL,
+                GENERATIVE_SERVICE_CONFIG,
+                GENERATIVE_TIMEOUT,
+                sending_variables,
+            )
+        except Exception as e:
+            sentry_sdk.capture_exception(e)
+            logger.exception(e)
+            hypotheses = []
+    else:
+        hypotheses = []
+    logger.info(f"generated hypotheses: {hypotheses}")
+
+    # if we do not have a goals from prompt, extract them using generative model (at most once in a dialog)
+    goals_from_prompt = ""
+    prompt_name = Path(PROMPT_FILE).stem
+    if not GOALS_FROM_PROMPT:
+        # get already collected goals from prompts from human attributes
+        prev_prompts_goals = int_ctx.get_prompts_goals(ctx, actor)
+        goals_from_prompt = prev_prompts_goals.get(prompt_name, "")
+        if not goals_from_prompt:
+            # if current prompt's goals are empty in human attributes, generate them!
+            goals_from_prompt = get_goals_from_prompt(
+                prompt=PROMPT,
+                url=GENERATIVE_SERVICE_URL,
+                generative_timeout=GENERATIVE_TIMEOUT,
+                sending_variables=sending_variables,
+            )
+            logger.info(f"Generated goals for prompt using generative service:\n{goals_from_prompt}")
+        else:
+            logger.info("Found goals for prompt from the human attributes")
+
+    for hyp in hypotheses:
+        confidence = DEFAULT_CONFIDENCE
+        if len(hyp) and hyp[-1] not in [".", "?", "!"]:
+            hyp += "."
+            confidence = LOW_CONFIDENCE
+        _curr_attrs = {"can_continue": CAN_NOT_CONTINUE}
+        if goals_from_prompt:
+            _curr_attrs["prompts_goals"] = {prompt_name: goals_from_prompt}
```

Comment:
you do not need anything related to prompt goals

Reply:
Done